PRINCIPLES.md

```diff
@@ -0,0 +1,254 @@
+
+3. [**The principle of continuous state reconciliation**](#3-continuous-state-reconciliation)
+
+    Software agents continuously, and automatically, compare a system's _Actual State_ to its _Desired State_.
+    If the actual and desired states differ, automated actions are immediately attempted to reconcile them.
+    These differences could be due to the actual state drifting from the desired state, or the desired state changing intentionally.
+
+4. [**The principle of operations through declaration**](#4-operations-through-declaration)
+
+    When wishing to operate on a software system, a human or software agent will not interact with the running system and modify it directly.
+    Instead, the agent will create a new declarative version of the desired state in the state store.
+
+## Introduction
+
+The software systems that we manage vary widely; from battery powered embedded systems driven by microcontrollers, to globally distributed applications with millions of users. 
+Most systems are composed of subsystems themselves made of software.
+It is impossible to comprehensively outline all the specific practices for managing such a variety of systems.
+
+However, despite the myriad differences, several important principles emerge that simplify the task of reliably managing and operating _all_ software systems at scale. GitOps attempts to capture some of these principles in a coherent framework.
+The GitOps principles can be applied to managing entire software systems or applied only to parts of larger systems.
+
+See also the [Rationale for GitOps](RATIONALE.md).
+
+## Scope
+
+GitOps concerns the verifiable behaviour of computer systems and their interfaces.
+Specifically, GitOps is _not_ about human processes, and is not intended as a model for judging human organisational designs and operational practices.
+
+The GitOps principles are to be used as guiding principles in the development of modern software and system operations. They do not form a concrete specification.
+
+The GitOps principles are a direction, not a destination. They should be applied pragmatically. For example, whilst desirable to apply them strictly to an entire software systems, they can also be applied loosely to selected parts of larger systems as part of a progressive adoption.
+
+## The GitOps Principles
+
+### 1. Declarative configuration
+
+<div style="background: #E3F2FD; border-left: 3px solid #0D47A1; padding:10px; margin-bottom:5px; font-style:italic">
+A system managed by GitOps must have its <em>Desired State</em> expressed declaritively as data.
+This declarative data must be in a format writable and readable by both humans and software.
+</div>
+
+#### What is a system's Desired State?
+
+_Configuration_ is a common feature of most software systems.
+By "Configuration", we mean _data that defines how a system or subsystem will behave and operate_.
+This configuration data is distinct and separate from the data a system will process.
```

Comment:
```suggestion
Configuration applies to the control surfaces of a system (i.e. control plane) 
```
I think the use of the work `data` is confusing - it has many different means depending on the context.

Reply:
I agree with the removal of `data`, I would simply use `Configuration`, the use of `data` seems redundant and confusing.

I don't agree with the statement that configuration applies to control surfaces of a system only. Configuration may directly determine the behavior of the software itself (think configuration parameters that define runtime characteristics of an application -ie. heap size, feature flags, etc- which should also be declared as configuration).

To me the fundamental difference is _configuration defines **how** the system will behave_, as opposed to _data which defines **what** the system will process_

Reply:
I think of control surfaces as the mechanism by which you apply the configuration to change behavior. 

How do we distinguish policy (give administrators admin access) from static state (list of users stored in git) and external state (pulling a list of users at reconcile stage) - Is it even necessary to do so?

Reply:
> reproducible

I would avoid this term as it has a very specific meaning in some contexts e.g. binary reproducibility - I suggest repeatable

Reply:
Removing that sentence does not diminish this section and makes it less ambiguous. I amended to that effect to prevent misunderstanding. The clarification is not really needed in this context anyway.